packaging/test_irods_client_rest_cpp.py

```diff
@@ -191,57 +191,64 @@ def test_put_configuration(self):
-                admin.assert_icommand(['icd', "/tempZone/home/rods/{}".format(coll)])
-                for do in dobjects:
-                    admin.assert_icommand(['itouch', do])
-                    path = "/tempZone/home/rods/{}/{}".format(coll, do)
-                    cmds = self.construct_meta_ops_for_target(path, "data_object")
-                    res = irods_rest.meta(token, cmds)
-                    self.assertEqual(
-                        json.loads(res)["op_status"],
-                        "SUCCESS"
-                    )
-
-
-
-            # create 5 resources
-            for resc in resources:
-                admin.assert_icommand(['iadmin', 'mkresc', resc, 'unixfilesystem', "localhost:/var/lib/storageVault{}".format(resc[-1])])
-                cmds = self.construct_meta_ops_for_target(resc, "resource")
-                res = irods_rest.meta(token, cmds)
-                self.assertEqual(
-                    json.loads(res)["op_status"],
-                    "SUCCESS"
-                )
-
+            path = pwd + '/' + data_object
+            admin.assert_icommand(['itouch', path])
+            cmds = construct_meta_ops_for_target(path, 'data_object')
+            res = irods_rest.meta(
+                token,
+                cmds
+            )
+            self.assertEqual(res, '')
+            admin.assert_icommand(['irm', path])
+
+    def test_meta_collection(self):
+        collection = 'collections'
+        token = irods_rest.authenticate('rods', 'rods', 'native')
+        pwd, _ lib.execute_command(['ipwd'])
+        with session.make_session_for_existing_admin() as admin:
+            path = pwd + '/' + collection
+            admin.assert_icommand(['imkdir', path])
+            cmds = construct_meta_ops_for_target(path, 'collection')
+            res = irods_rest.meta(
+                token,
+                cmds
+            )
+            self.assertEqual(res, '')
+            admin.assert_icommand(['irmdir', path])
+
+    def test_meta_user(self):
+        user = 'user'
+        token = irods_rest.authenticate('rods', 'rods', 'native')
+        pwd, _ lib.execute_command(['ipwd'])
+        with session.make_session_for_existing_admin() as admin:
+            path = pwd + '/' + collection
+            admin.assert_icommand(['iadmin', 'mkuser', user, 'rodsuser'])
+            cmds = construct_meta_ops_for_target(user, user)
+            res = irods_rest.meta(
+                token,
+                cmds
+            )
+            self.assertEqual(res, '')
+            admin.assert_icommand('iadmin', 'rmuser', user)
+    def test_meta_resource(self):
+        resource = 'resource'
+        token = irods_rest.authenticate('rods', 'rods', 'native')
+        pwd, _ lib.execute_command(['ipwd'])
+        with session.make_session_for_existing_admin() as admin:
+            admin.assert_icommand(['iadmin', 'mkresc', resource, 'unixfilesystem', '`hostname`:/var/lib/irods/new_vault'])
```

Comment:
i think we have a `lib` call for this...

Reply:
Yes, `create_ufs_resource`. I hate that we put "user" (aka session) as the second parameter rather than the first, but other than that it sure is a lot less typing/reading :p